transformer/Transform.go

```diff
@@ -0,0 +1,228 @@
+package transformer
+
+import (
+	"bufio"
+	"strings"
+	"fmt"
+	"os"
+	"io/ioutil"
+	"gopkg.in/yaml.v2"
+	"log"
+
+	"github.com/cloud66/starter/common"
+)
+
+//main transformation format function
+func Transformer(filename string, formatTarget string, directlyTransformed bool) (bool, error) {
+
+	var err error
+	_, err = os.Stat(formatTarget)
+
+	// create file if not exists
+	if os.IsNotExist(err) {
+		var file, err = os.Create(formatTarget)
+		checkError(err)
+		defer file.Close()
+	} else {
+		common.PrintError("File %s already exists. Will be overwritten.\n", formatTarget)
+	}
+
+
+	yamlFile, err := ioutil.ReadFile(filename)
+
+	dockerCompose := Docker_compose{
+		Services: make(map[string]docker_Service),
+		Version:  "",
+	}
+
+	serviceYaml := Serviceyml{
+		Services: make(map[string]common.Service),
+		Dbs:      make([]string, 0),
+	}
+
+	if err := yaml.Unmarshal([]byte(yamlFile), &dockerCompose); err != nil {
+		fmt.Println(err.Error())
+	}
+
+	if len(dockerCompose.Services) == 0 {
+		d := make(map[string]docker_Service)
+		err = yaml.Unmarshal([]byte(yamlFile), &d)
+		checkError(err)
+
+		serviceYaml.Services, serviceYaml.Dbs = copyToServiceYML(d, directlyTransformed)
+
+	} else {
+
+		serviceYaml.Services, serviceYaml.Dbs = copyToServiceYML(dockerCompose.Services, directlyTransformed)
+	}
+	if len(serviceYaml.Dbs) != 0 {
```

Comment:
Cannot reproduce anymore. Fixed already, code was not doing anything.

Solved. 👍